### DISCUSSION TITLE
Incorrect decomposition by cvxEDA

### DISCUSSION THREAD

**Discussion by JiaerCGuo** [neuropsychology/NeuroKit#1047] [Q&A]
Hello all, 

I'm new to Neurokit2 and cvxEDA so I would really appreciate some help here. I'm running `eda_process` on an EDA data downsampled by a factor of 20 from 500Hz to 25Hz. Here is the code I'm using to process:

```
signals, info = nk.eda_process(data_ds_event['EDA'], sampling_rate=25,
                              method_cleaning='neurokit',
                              method_phasic='cvxeda',
                              method_peaks='nabian2018',
                              report = 'text')
```

When I put down `sampling_rate=25`, the cvxEDA method's decomposition seemed incorrect 
![image](https://github.com/user-attachments/assets/b27dc0dd-5871-46cc-a1f4-9063ba0d8428)

If I changed it to `sampling_rate=10`, the cvxEDA decomposition became better
![image](https://github.com/user-attachments/assets/f80c37b3-cf8f-4eed-8216-1b85b62970b7)

When I changed the decomposition method to `highpass` and kept the `sampling_rate=25`, the results seemed fine, so I'm assuming the problem is not on the dataset itself.
![image](https://github.com/user-attachments/assets/c0e3903f-3775-4463-8277-db1fa166f731)

Does anyone know what might be the problem here? I also tried to downsample the data to 10Hz, but similar thing happened that cvxEDA doesn't work well with the correct sampling rate, but better when the defined rate is lower. 

Attached is the data I'm working with in case it's helpful to troubleshoot. Thanks in advance for any insights into this issue!

Cecilia Guo

Update: file with correct time stamps
[data_ds_event.csv](https://github.com/user-attachments/files/17652198/data_ds_event.csv)

**Comment by DominiqueMakowski** [comment]
> the cvxEDA method's decomposition seemed incorrect

What do you mean by that, it looks pretty alright to me

**Reply by JiaerCGuo** [reply]
This might be clearer with the overlapped raw and tonic components. The first half of the tonic seems poorly decomposed with peaks and troughs that shouldn't be there. 
![image](https://github.com/user-attachments/assets/48c6259e-e405-49c6-a00f-54897c0fe413)

**Reply by DominiqueMakowski** [reply]
the tonic peaks are fairly low frequency, it does not look *that* bad IMO (but then I'm mostly interested in the phasic signal, not too sure when it becomes a really good tonic)
